code/modules/mob/living/silicon/mommi/mommi_subtypes/sammi.dm

```diff
@@ -0,0 +1,211 @@
+				user.visible_message("<span class='notice'>[user.name] enters commands into [src.name].</span>")
+			} else {
+				to_chat(user, "The console's cover is closed.")
+			}
+
+	else if(W.is_screwdriver(user) && opened && !cell)	// haxing
+		wiresexposed = !wiresexposed
+		to_chat(user, "The wires have been [wiresexposed ? "exposed" : "unexposed"].")
+		updateicon()
+
+	else if(W.is_screwdriver(user) && opened && cell)	// radio
+		if(radio)
+			radio.attackby(W,user)//Push it to the radio to let it handle everything
+		else
+			to_chat(user, "Unable to locate a radio.")
+		updateicon()
+
+	else if(istype(W, /obj/item/device/encryptionkey/) && opened)
+		if(radio)//sanityyyyyy
+			radio.attackby(W,user)//GTFO, you have your own procs
+		else
+			to_chat(user, "Unable to locate a radio.")
+/*
+	else if (istype(W, /obj/item/weapon/card/id)||istype(W, /obj/item/device/pda))			// trying to unlock the interface with an ID card
+		if(emagged)//still allow them to open the cover
+			to_chat(user, "The interface seems slightly damaged")
+		if(opened)
+			to_chat(user, "You must close the cover to swipe an ID card.")
+		else
+			if(allowed(usr))
+				locked = !locked
+				to_chat(user, "You [ locked ? "lock" : "unlock"] [src]'s interface.")
+				updateicon()
+			else
+				to_chat(user, "<span class='warning'>Access denied.</span>")
+*/
+
+	else if(istype(W, /obj/item/weapon/wrench)) // Need to make this not bludgeon them
+		if(anchored)
+			to_chat(user, "<span class='notice'>You unbolt the SAMMI from the floor.</span>")
+			anchored = 0
+			updateicon()
+
+		else
+			to_chat(user, "<span class='notice'>You anchor the SAMMI to the floor.</span>")
+			anchored = 1
+			updateicon()
+
+		return 0
+
+
+	else if(istype(W, /obj/item/borg/upgrade/))
+		var/obj/item/borg/upgrade/U = W
+		U.attempt_action(src,user)
+
+	else if(istype(W, /obj/item/device/camera_bug))
+		help_shake_act(user)
+		return 0
+
+	else
+		//spark_system.start()
+		return ..()
+
+/mob/living/silicon/robot/mommi/sammi/New(loc)
+	..()
+	module = new /obj/item/weapon/robot_module/mommi/sammi(src)
+	if(emagged){
+		laws = new sammi_base_law_type
+	} else {
+		laws = new sammiemag_base_law_type
+	}
+/mob/living/silicon/robot/mommi/sammi/proc/transfer_personality(var/client/candidate)
+
+	if(!candidate)
+		return
+
+	src.mind = candidate.mob.mind
+	src.ckey = candidate.ckey
+	if(src.mind)
+		src.mind.assigned_role = "SAMMI"
+
+/mob/living/silicon/robot/mommi/sammi/ghost()
+	//if(src.subtype == "sammi")
+	if(client && key)
+		ghostize(1)
+		src.mind.current = src.mind.original
+		src.visible_message("<span class=\"warning\">[src] disconnects from the network...attempting to reconnect!</span>")
+		icon_state = "sammi_offline"
+		updateicon()
+
+/mob/living/silicon/robot/mommi/sammi/attack_ghost(var/mob/dead/observer/O)
```

Comment:
Why not use the transmog mechanic that ShadowMech coded ?
It's used for midround bombermen ghosts (who retain their identity) and holomen at the holodeck.